src/aquarium.py

```diff
@@ -40,119 +43,140 @@
-
-
-@app.on_event("shutdown")  # type: ignore
-async def on_shutdown():
-    logger.info("Aquarium shutdown!")
-    await api.state.gstate.shutdown()
-    logger.info("shutting down node manager")
-    await api.state.nodemgr.shutdown()
-
-
-api.include_router(local.router)
-api.include_router(bootstrap.router)
-api.include_router(orch.router)
-api.include_router(status.router)
-api.include_router(services.router)
-api.include_router(nodes.router)
-api.include_router(devices.router)
-api.include_router(nfs.router)
-
-
-#
-# mounts
-#   these should be the last things to be done, so fastapi is aware of
-#   everything that comes before.
-#
-# api calls go here.
-app.mount(
-    "/api",
-    api,
-    name="api"
-)
-# mounting root "/" must be the last thing, so it does not override "/api".
-app.mount(
-    "/",
-    StaticFiles(directory="./glass/dist/", html=True),
-    name="static"
-)
+def app_factory():
+    logger: logging.Logger = fastapi_logger
+
+    api_tags_metadata = [
+        {
+            "name": "local",
+            "description": "Operations local to the node where the endpoint is being invoked."
+        },
+        {
+            "name": "bootstrap",
+            "description": "Allows creating a minimal cluster on the node."
+        },
+        {
+            "name": "orch",
+            "description": "Operations related to Ceph cluster orchestration."
+        },
+        {
+            "name": "status",
+            "description": "Allows obtaining operation status information."
+        },
+        {
+            "name": "services",
+            "description": "Create, destroy, and operate Aquarium Services."
+        },
+        {
+            "name": "nodes",
+            "description": "Perform Aquarium Cluster node operations"
+        },
+        {
+            "name": "devices",
+            "description": "Obtain and perform operations on cluster devices"
+        }
+    ]
+
+    aquarium_app = FastAPI(docs_url=None)
+    aquarium_api = FastAPI(
+        title="Project Aquarium",
+        description="Project Aquarium is a SUSE-sponsored open source project " +
+                    "aiming at becoming an easy to use, rock solid storage " +
+                    "appliance based on Ceph.",
+        version="1.0.0",
+        openapi_tags=api_tags_metadata)
+
+    @aquarium_app.on_event("startup")  # type: ignore
+    async def on_startup():
+
+        lvl = "INFO" if not os.getenv("AQUARIUM_DEBUG") else "DEBUG"
+        setup_logging(lvl)
+        logger.info("Aquarium startup!")
+
+        gstate: GlobalState = GlobalState()
+
+        # init node mgr
+        logger.info("starting node manager")
+        nodemgr: NodeMgr = NodeMgr(gstate)
+
+        # Set up Ceph connections
+        ceph: Ceph = Ceph()
+        ceph_mgr: Mgr = Mgr(ceph)
+        gstate.add_ceph_mgr(ceph_mgr)
+        ceph_mon: Mon = Mon(ceph)
+        gstate.add_ceph_mon(ceph_mon)
+        cephfs: CephFS = CephFS(ceph_mgr, ceph_mon)
```

Comment:
I don't see a good reason to keep track of a cephfs instance. These should not be as ubiquitous as the mon's and the mgr's, and I think creating them just where/when they're needed would be fine.